src/app/components/VisuallyHiddenText/index.tsx

```diff
@@ -7,17 +7,26 @@ interface VisuallyHiddenTextProps {
   id?: string;
   tabIndex?: number;
   as?: React.ElementType;
+  lang?: string;
 }
 
 const VisuallyHiddenText = ({
   children,
   id,
   tabIndex,
   as,
+  lang,
+  ...htmlAttributes
```

Comment:
In the component standards we want to typically avoid props spreading. However, we are not sure of an alternative way we could allow the `data-testid` to pass through successfully in this example. Open to suggestions if we are not willing to proceed with this solution.

Reply:
May be worth taking inspiration from the Text component for type-checking the `htmlAttributes`:
https://github.com/bbc/simorgh/blob/latest/src/app/components/Text/index.tsx#L9-L24

It uses the `T` generic to get the HTML types from the passed in `as` prop. This means that if you set `as="a"` to set it to an anchor tag for example, it would then allow you to specify `href=""` as a prop. Whereas if you set `as="div"` and tried to set `href=""`, you would get a type error as `href` isn't a valid attribute on a `div` element.

Reply:
i've updated the structure of the `VHT` but `htmlAttributes` still needs to be spread on the component to provide the `data-testid` unless i'm implementing the type-checking incorrectly? Would you mind taking another peek at the updated code?

Reply:
I think it may need this line too:
```
VisuallyHiddenTextProps<T> &
  Omit<ComponentPropsWithoutRef<T>, keyof VisuallyHiddenTextProps<T>>)
```

This should add some type-checking to the remaining spread `htmlAttributes`. The `ComponentPropsWithoutRef` type seems to contain all the types for HTML elements

Reply:
Could another option be to get the VisuallyHiddenText component to call the Text component - which has all of this logic? It could reduce the complexity / duplication?

Reply:
That was the original implementation of the `VisuallyHiddenText`, but the problem is the `Text` component is quite opinionated about things like styles, font sizes, font spacings etc which can't not be set. 
https://github.com/bbc/simorgh/blob/153de052bf04d31a96532144e41316de63a0c1b0/src/app/components/Text/index.tsx#L29-L35

So to maintain as close parity as possible to old `VisuallyHiddenText` component, it was opted not to use it.

Reply:
Ah of course, makes sense. Thanks for explaining.